validator/client/src/test/java/tech/pegasys/teku/validator/client/RuntimeProposerConfigTest.java

```diff
@@ -37,19 +38,57 @@ class RuntimeProposerConfigTest {
   private final Eth1Address address = dataStructureUtil.randomEth1Address();
   private final Eth1Address address2 = dataStructureUtil.randomEth1Address();
 
+  private final UInt64 gasLimit = dataStructureUtil.randomUInt64();
+  private final UInt64 gasLimit2 = dataStructureUtil.randomUInt64();
+
   @Test
   void shouldAddEntries() {
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).isEmpty();
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).isEmpty();
     proposerConfig.addOrUpdate(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
+    proposerConfig.addOrUpdate(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+  }
+
+  @Test
+  void shouldKeepExistingFeeRecipient() {
+    assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).isEmpty();
```

Comment:
we shouldn't need to test up front that things are clear, this should be consistent each test.. maybe 1 test to show thats true, then the rest can remove, otherwise it generates a lot of noise.